ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -347,11 +346,16 @@ DNSDaemonListener createDaemonListener() {
                 .build();
         final DiscoveryPeer peer = DiscoveryPeer.fromEnode(enodeURL);
         peers.add(peer);
-        rlpxAgent.connect(peer);
       }
-      // only replace dnsPeers if the lookup was successful:
       if (!peers.isEmpty()) {
-        dnsPeers.set(peers);
+        final Optional<PeerDiscoveryController> peerDiscoveryController =
+            peerDiscoveryAgent.getPeerDiscoveryController();
+        if (peerDiscoveryController.isPresent()) {
+          final PeerDiscoveryController controller = peerDiscoveryController.get();
+          LOG.debug("Adding {} DNS peers to PeerTable", peers.size());
+          peers.forEach(controller::addToPeerTable);
```

Comment:
As mentioned previously, we should be calling `PeerDiscoveryAgent.bond` here instead of directly injecting peers into the discovery peer table.

Reply:
Yes!